src/controller/python/chip/clusters/Attribute.py

```diff
@@ -65,17 +78,92 @@ def __hash__(self):
         return str(self).__hash__()
 
 
+@dataclass
+class EventPath:
+    EndpointId: int = None
+    ClusterId: int = None
+    EventId: int = None
+
+    def __init__(self, EndpointId: int = None, Cluster=None, Event=None, ClusterId=None, EventId=None):
+        self.EndpointId = EndpointId
+        if Cluster is not None:
+            # Wildcard read for a specific cluster
+            if (Event is not None) or (ClusterId is not None) or (EventId is not None):
+                raise Warning(
+                    "Event, ClusterId and AttributeId is ignored when Cluster is specified")
+            self.ClusterId = Cluster.id
+            return
+        if Event is not None:
+            if (ClusterId is not None) or (EventId is not None):
+                raise Warning(
+                    "ClusterId and EventId is ignored when Event is specified")
+            self.ClusterId = Event.cluster_id
+            self.EventId = Event.event_id
+            return
+        self.ClusterId = ClusterId
+        self.EventId = EventId
+
+    def __str__(self) -> str:
+        return f"{self.EndpointId}/{self.EventId}/{self.EventId}"
+
+    def __hash__(self):
+        return str(self).__hash__()
+
+
 @dataclass
 class AttributePathWithListIndex(AttributePath):
     ListIndex: int = None
 
 
+@dataclass
+class EventHeader:
+    EndpointId: int = None
+    ClusterId: int = None
+    EventId: int = None
+    EventNumber: int = None
+    Priority: EventPriority = None
+    Timestamp: int = None
+    TimestampType: EventTimestampType = None
+
+    def __init__(self, EndpointId: int = None, Cluster=None, Event=None, ClusterId=None, EventId=None, EventNumber=None, Priority=None, Timestamp=None, TimestampType=None):
+        self.EndpointId = EndpointId
+        if Cluster is not None:
+            # Wildcard read for a specific cluster
```

Comment:
Why is anything about wildcards even mentioned here? Isn't this supposed to be paths in event headers, which are always concrete?

Reply:
thanks for catching up this issue, yes, removed and updated, thanks